File: php/annalyns-infiltration/AnnalynsInfiltration.php
<?php

class AnnalynsInfiltration
{
    public function canFastAttack($is_knight_awake)
    {
        if (!$is_knight_awake)
            return true;
        else
            return false;
    }

    public function canSpy(
        $is_knight_awake,
        $is_archer_awake,
        $is_prisoner_awake
    ) {
        if ($is_knight_awake || $is_archer_awake || $is_prisoner_awake)
            return true;
        else
            return false;
    }

    public function canSignal(
        $is_archer_awake,
        $is_prisoner_awake
    ) {
        if (!$is_archer_awake && $is_prisoner_awake)
            return true;
        else
            return false;
    }

    public function canLiberate(
        $is_knight_awake,
        $is_archer_awake,
        $is_prisoner_awake,
        $is_dog_present
    ) {
        if($is_knight_awake && !$is_archer_awake && $is_prisoner_awake && $is_dog_present)
            return true;
        else if ($is_knight_awake && !$is_archer_awake && !$is_prisoner_awake && $is_dog_present)
            return true;
        else if (!$is_knight_awake && !$is_archer_awake && ($is_prisoner_awake || $is_dog_present))
            return true;
        else
            return false;
    }
}

File: java/annalyns-infiltration/src/main/java/AnnalynsInfiltration.java
class AnnalynsInfiltration {
    public static boolean canFastAttack(boolean knightIsAwake) {
        if (!knightIsAwake)
            return true;
        else
            return false;
    }

    public static boolean canSpy(boolean knightIsAwake, boolean archerIsAwake, boolean prisonerIsAwake) {
        if (knightIsAwake || archerIsAwake || prisonerIsAwake)
            return true;
        else
            return false;
    }

    public static boolean canSignalPrisoner(boolean archerIsAwake, boolean prisonerIsAwake) {
         if (!archerIsAwake && prisonerIsAwake)
            return true;
        else
            return false;
    }

    public static boolean canFreePrisoner(boolean knightIsAwake, boolean archerIsAwake, boolean prisonerIsAwake, boolean petDogIsPresent) {
        if ((prisonerIsAwake && !knightIsAwake && !archerIsAwake) || (petDogIsPresent && !archerIsAwake))
            return true;
        else
            return false;
    }
}
